File: tests/helpers/fixtures.ts
import type {
  Company,
  Contractor,
  Invoice,
  JobDescription,
  MyCompanyInfo,
  Property,
  Supervisor,
} from "../../src/types";

export function makeMyCompanyInfo(over: Partial<MyCompanyInfo> = {}): MyCompanyInfo {
  return {
    id: 1,
    name: "DKSK Official",
    phone: "555-0100",
    email: "ops@dksk.example",
    address: "1 Paint St",
    zip: "12345",
    license_number: "LIC-001",
    ...over,
  };
}

export function makeProperty(over: Partial<Property> = {}): Property {
  return {
    id: 100,
    name: "Main Unit",
    address: "123 Main St",
    city: "Hilo",
    zip: "96720",
    gate_code: "1234",
    garage_remote_code: null,
    lock_box: "A1B2",
    special_note: null,
    manager_name: null,
    manager_phone: null,
    manager_email: null,
    is_active: true,
    supervisor_id: 10,
    ...over,
  };
}

export function makeSupervisor(over: Partial<Supervisor> = {}): Supervisor {
  return {
    id: 10,
    name: "Pat Supervisor",
    phone: null,
    email: null,
    company_id: 1,
    properties: [makeProperty()],
    ...over,
  };
}

export function makeCompany(over: Partial<Company> = {}): Company {
  return {
    id: 1,
    company_id: 1001,
    name: "Acme Properties",
    owner: "Jane",
    phone: null,
    email: null,
    address: null,
    city: null,
    zip: null,
    special_note: null,
    supervisors: [makeSupervisor()],
    ...over,
  };
}

export function makeContractor(over: Partial<Contractor> = {}): Contractor {
  return {
    id: 1,
    name: "Alex Painter",
    license_number: "C-1",
    social_security_number: null,
    contractor_id: null,
    payroll_percent: "10",
    cell_phone: null,
    email: null,
    address: null,
    city: null,
    zip: null,
    special_note: null,
    is_active: true,
    ...over,
  };
}

export function makeJob(over: Partial<JobDescription> = {}): JobDescription {
  return {
    id: 1,
    description: "Interior paint",
    size_bedroom: 2,
    size_bathroom: 2,
    price: 100,
    ...over,
  };
}

export function makeInvoice(over: Partial<Invoice> = {}): Invoice {
  return {
    id: 1,
    todays_date: "2026-05-25",
    work_date: "2026-05-20",
    company_name: "Acme Properties",
    property_address: "123 Main St",
    unit: "A",
    gate_code: null,
    lock_box: null,
    size_bedroom: 2,
    size_bathroom: 2,
    work_order: null,
    job_description_choice: JSON.stringify(["Interior paint"]),
    contractor_name: "Alex Painter",
    amount_cost: 100,
    amount_paid1: 0,
    date_paid1: null,
    check_number1: null,
    amount_paid2: 0,
    date_paid2: null,
    check_number2: null,
    invoice_created_date: null,
    special_note: null,
    garage_remote_code: null,
    status: 0,
    ...over,
  };
}

File: src-tauri/tests/common/fixtures.rs
//! Reusable builders for Rust integration tests. Not all builders are used by
//! every test file; suppress the dead_code lint that fires for unused
//! per-file helpers.

#![allow(dead_code)]

use paint_contractor_lib::models::*;

pub fn make_invoice() -> Invoice {
    Invoice {
        id: 0,
        todays_date: "2026-05-25".into(),
        work_date: "2026-05-20".into(),
        company_name: "Acme".into(),
        property_address: "1 Main".into(),
        unit: "A".into(),
        gate_code: None,
        lock_box: None,
        size_bedroom: 2,
        size_bathroom: 2,
        work_order: Some("WO-1".into()),
        job_description_choice: "[\"Paint\"]".into(),
        contractor_name: "Alex".into(),
        amount_cost: 100,
        amount_paid1: 0,
        date_paid1: None,
        check_number1: None,
        amount_paid2: 0,
        date_paid2: None,
        check_number2: None,
        invoice_created_date: Some("2026-05-25".into()),
        special_note: None,
        garage_remote_code: None,
        status: 0,
    }
}

pub fn make_company(name: &str, company_id: i32) -> Company {
    Company {
        id: 0,
        company_id,
        name: name.into(),
        owner: None,
        phone: None,
        email: None,
        address: None,
        city: None,
        zip: None,
        special_note: None,
        supervisors: vec![],
    }
}

pub fn make_supervisor(name: &str) -> Supervisor {
    Supervisor {
        id: 0,
        name: name.into(),
        phone: None,
        email: None,
        company_id: 0,
        properties: vec![],
    }
}

pub fn make_property(name: &str) -> Property {
    Property {
        id: 0,
        name: name.into(),
        address: Some(format!("addr-{name}")),
        city: None,
        zip: None,
        gate_code: None,
        garage_remote_code: None,
        lock_box: None,
        special_note: None,
        manager_name: None,
        manager_phone: None,
        manager_email: None,
        is_active: Some(true),
        supervisor_id: 0,
    }
}

pub fn make_contractor(name: &str) -> Contractor {
    Contractor {
        id: 0,
        name: name.into(),
        license_number: None,
        social_security_number: None,
        contractor_id: None,
        payroll_percent: None,
        cell_phone: None,
        email: None,
        address: None,
        city: None,
        zip: None,
        special_note: None,
        is_active: Some(true),
    }
}
